sklearn/datasets/openml.py

```diff
@@ -0,0 +1,335 @@
+        else:
+            return None
+    json_data = json.loads(response.read().decode("utf-8"))
+    response.close()
+    return json_data
+
+
+def _convert_arff_data(arff_data):
+    """
+    converts the arff object into the appropriate matrix type (now: np.ndarray,
+    later also: scipy.sparse.csr_matrix) based on the 'data part' (i.e., in the
+    liac-arff dict, the object from the 'data' key)
+
+    Parameters
+    ----------
+    arff_data : list or dict
+        as obtained from liac-arff object
+
+    Returns
+    -------
+    X : np.ndarray
+        (or later also: scipy.sparse.csr_matrix)
+    """
+    if isinstance(arff_data, list):
+        X = np.array(arff_data, dtype=object)
+    # elif: extendable for sparse arff in future ()
+    else:
+        # This should never happen
+        raise ValueError('Unexpected Data Type obtained from arff.')
+    return X
+
+
+def _get_data_info_by_name(name, version):
+    """
+    Utilizes the openml dataset listing api to find a dataset by
+    name/version
+    OpenML api fn:
+    https://www.openml.org/api_docs#!/data/get_data_list_data_name_data_name
+
+    Parameters
+    ----------
+    name : str
+        name of the dataset
+
+    version : int or str
+        If version is an integer, the exact name/version will be obtained from
+        OpenML. If version is a string (value: "active") it will take the first
+        version from OpenML that is annotated as active. Any other string
+        values except "active" are treated as integer.
+
+    Returns
+    -------
+    json_data['data']['dataset'][0]: json
+        json representation of the first dataset object that adhired to the
+        search criteria
+
+    """
+    if version == "active":
+        # situation in which we return the oldest active version
+        url = _SEARCH_NAME.format(name) + "/status/active/"
+        error_msg = "No active dataset {} found.".format(name)
+        json_data = _get_json_content_from_openml_api(url, error_msg, True)
+        return json_data['data']['dataset'][0]
+
+    # an integer version has been provided
+    url = (_SEARCH_NAME + "/data_version/{}").format(name, version)
+    json_data = _get_json_content_from_openml_api(url, None, False)
+    if json_data is None:
+        # we can do this in 1 fn call if OpenML does not require the
+        # specification of the dataset status (i.e., return datasets with a
+        # given name / version regardless of active, deactivated, etc. )
+        # TODO: feature request OpenML.
+        url += "/status/deactivated"
+        error_msg = "Dataset {} with version {} not found.".format(name,
+                                                                   version)
+        json_data = _get_json_content_from_openml_api(url, error_msg, True)
+
+    return json_data['data']['dataset'][0]
+
+
+def _get_data_description_by_id(data_id):
+    # OpenML API fn: https://www.openml.org/api_docs#!/data/get_data_id
```

Comment:
remove "fn" or replace it by "url". Same below

Reply:
Agreed